dashboard/app/reporting.go

```diff
@@ -227,9 +228,43 @@ func handleReportNotif(c context.Context, typ string, bug *Bug) (*dashapi.BugNot
 		commits := strings.Join(bug.Commits, "\n")
 		return createNotification(c, dashapi.BugNotifBadCommit, true, commits, bug, reporting, bugReporting)
 	}
+	for _, label := range bug.Labels {
+		if label.SetBy != "" {
+			continue
+		}
+		str := label.String()
+		if reporting.Labels[str] == "" {
+			continue
+		}
+		if stringInList(bugReporting.GetLabels(), str) {
+			continue
+		}
+		return createLabelNotification(c, label, bug, reporting, bugReporting)
+	}
 	return nil, nil
 }
 
+func createLabelNotification(c context.Context, label BugLabel, bug *Bug, reporting *Reporting,
```

Comment:
Can these labels exist when we first report a bug? We generally include all we have when first reporting the bug (reproducers, bisections). Should we include labels there as well?

Reply:
Trying to imagine how a report could look like then.

```
Hello,

syzbot found the following issue on:

HEAD commit:    101010101010 a really long title, longer than 80 chars, re..
git tree:       repo10alias
console output: %[3]v
kernel config:  %[4]v
dashboard link: https://testapp.appspot.com/bug?extid=%[1]v
compiler:       compiler10
userspace arch: i386
syz repro:      %[2]v
CC:             [bar@foo.com foo@bar.com maintainers@repo10.org bugs@repo10.org]

IMPORTANT: if you fix the issue, please add the following tag to the commit:
Reported-by: syzbot+%[1]v@testapp.appspotmail.com

report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 


Bug presence analysis results: the bug reproduces only on the downstream tree.

syzbot has run the reproducer on other relevant kernel trees and got
the following results:

downstream (commit 947548860) on 2000/01/11:
crash title
Report: %URL%

lts (commit 947548860) on 2000/01/11:
Didn't crash.
upstream (commit 947548860) on 2000/01/11:

Didn't crash.
More details can be found at:
%URL%

---
This report is generated by a bot. It may contain errors.
See https://goo.gl/tpsmEJ for more information about syzbot.
syzbot engineers can be reached at syzkaller@googlegroups.com.

syzbot will keep track of this issue. See:
https://goo.gl/tpsmEJ#status for how to communicate with syzbot.

If the bug is already fixed, let syzbot know by replying with:
#syz fix: exact-commit-title

If you want syzbot to run the reproducer, reply with:
#syz test: git://repo/address.git branch-or-commit-hash
If you attach or paste a git patch, syzbot will apply it before testing.

If you want to change bug's subsystems, reply with:
#syz set subsystems: new-subsystem
(See the list of subsystem names on the web dashboard)

If the bug is a duplicate of another bug, reply with:
#syz dup: exact-subject-of-another-report

If you want to undo deduplication, reply with:
#syz undup
```

Isn't it becoming too big?

Reply:
If we add it just as e.g.

```

HEAD commit:    101010101010 a really long title, longer than 80 chars, re..
git tree:       repo10alias
console output: %[3]v
kernel config:  %[4]v
dashboard link: https://testapp.appspot.com/bug?extid=%[1]v
compiler:       compiler10
userspace arch: i386
syz repro:      %[2]v
CC:             [bar@foo.com foo@bar.com maintainers@repo10.org bugs@repo10.org]
Labels:         origin:downstream, prio:high
```

then it won't be noticeable.

Reply:
A middle ground would be to add a separate line:
```
Bug presence analysis results: the bug reproduces only on the downstream tree.
```
but without all the details.
But it's better to place it above the report then, report can be long.

Reply:
Something like this?
Though I think in the end we'd barely ever face this use case on syzbot.

```
Hello,

syzbot found the following issue on:

HEAD commit:    101010101010 a really long title, longer than 80 chars, re..
git tree:       repo10alias
console output: %[3]v
kernel config:  %[4]v
dashboard link: https://testapp.appspot.com/bug?extid=%[1]v
compiler:       compiler10
userspace arch: i386
syz repro:      %[2]v
CC:             [bar@foo.com foo@bar.com maintainers@repo10.org bugs@repo10.org]

Bug presence analysis results: the bug reproduces only on the downstream tree.

IMPORTANT: if you fix the issue, please add the following tag to the commit:
Reported-by: syzbot+%[1]v@testapp.appspotmail.com

report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 
report1 report1 report1 report1 report1 report1 report1 report1 

---
This report is generated by a bot. It may contain errors.
See https://goo.gl/tpsmEJ for more information about syzbot.
syzbot engineers can be reached at syzkaller@googlegroups.com.

syzbot will keep track of this issue. See:
https://goo.gl/tpsmEJ#status for how to communicate with syzbot.

If the bug is already fixed, let syzbot know by replying with:
#syz fix: exact-commit-title

If you want syzbot to run the reproducer, reply with:
#syz test: git://repo/address.git branch-or-commit-hash
If you attach or paste a git patch, syzbot will apply it before testing.

If you want to change bug's subsystems, reply with:
#syz set subsystems: new-subsystem
(See the list of subsystem names on the web dashboard)

If the bug is a duplicate of another bug, reply with:
#syz dup: exact-subject-of-another-report

If you want to undo deduplication, reply with:
#syz undup
```